src/test/resources/pages/adminAccountDetails.html

```diff
@@ -1,14 +1,16 @@
 <html>
 <head>
-<title>
-  Instructor Account Details [Administrator]
-</title>
-<link href="/favicon.png" rel="shortcut icon">
-<meta charset="utf-8">
-<meta content="width=device-width, initial-scale=1.0" name="viewport">
-<link href="https://cdn.jsdelivr.net/npm/bootstrap@3.1.1/dist/css/bootstrap.min.css" rel="stylesheet" type="text/css">
-<link href="https://cdn.jsdelivr.net/npm/bootstrap@3.1.1/dist/css/bootstrap-theme.min.css" rel="stylesheet" type="text/css">
-<link href="/stylesheets/teammatesCommon.css" rel="stylesheet" type="text/css">
+  <title>
+    Instructor Account Details [Administrator]
+  </title>
+  <link href="/favicon.png" rel="shortcut icon">
+  <meta charset="utf-8">
+  <meta http-equiv="Content-Security-Policy" content="default-src 'self';
+        script-src 'self' https://cdn.jsdelivr.com; style-src 'self' https://cdn.jsdelivr.com 'nonce-2726c7f26a'">
```

Comment:
hard-coding 2726c7f26a is not really the right way to use a nonce. A nonce is supposed to be random. But is the nonce really needed here though?

Reply:
I think yes because we have inline css.

Reply:
I think the right way would be to add that css style rule into a css file instead of keeping the inline style.

Also, I think it's better to use `filter` in the `web.xml` file to add custom headers in our HTTP response instead of modifying all the HTML and JSP files. It's a good opportunity to learn more about how a Java Web app works too 😄 .

>Is there anything else I have to change for the csp to be rightly implemented ?

I don't know exactly what the "right implementation of CSP" is. It's a new thing that we are trying to add to TEAMMATES. But that's also exactly why your help would be very valuable, if you choose to undertake this task. 

You can think of this task more like self-studying about a topic and then propose changes for discussion. I will use my knowledge and experience to comment on what you propose, so that hopefully in the end we arrive at the "right implementation". Thus, you should try to explain the reasoning behind what you do as much as possible, e.g. why you are adding a nonce etc.

This process maybe slow as it would require quite a few rounds of interactions between us, so I hope you are prepared for it, if you decide to continue with this task.